Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.h

```diff
@@ -30,42 +30,43 @@
 #include <WebCore/PlatformSpeechSynthesisUtterance.h>
 #include <WebCore/PlatformSpeechSynthesisVoice.h>
 #include <WebCore/SpeechSynthesisClient.h>
+#include <wtf/RefCounted.h>
 #include <wtf/TZoneMalloc.h>
 
-namespace WebKit {
-class WebSpeechSynthesisClient;
-}
-
-namespace WTF {
-template<typename T> struct IsDeprecatedWeakRefSmartPointerException;
-template<> struct IsDeprecatedWeakRefSmartPointerException<WebKit::WebSpeechSynthesisClient> : std::true_type { };
-}
-
 namespace WebKit {
 
 class WebPage;
-    
-class WebSpeechSynthesisClient : public WebCore::SpeechSynthesisClient {
+
+class WebSpeechSynthesisClient : public RefCounted<WebSpeechSynthesisClient>, public WebCore::SpeechSynthesisClient {
     WTF_MAKE_TZONE_ALLOCATED(WebSpeechSynthesisClient);
 public:
-    explicit WebSpeechSynthesisClient(WebPage&);
+    static Ref<WebSpeechSynthesisClient> create(WebPage& page)
+    {
+        return adoptRef(*new WebSpeechSynthesisClient(page));
+    }
+
     virtual ~WebSpeechSynthesisClient() { }
-    
+
+    void ref() const final { RefCounted::ref(); }
+    void deref() const final { RefCounted::deref(); }
+
     const Vector<RefPtr<WebCore::PlatformSpeechSynthesisVoice>>& voiceList() override;
     void speak(RefPtr<WebCore::PlatformSpeechSynthesisUtterance>) override;
     void cancel() override;
     void pause() override;
     void resume() override;
 
 private:
+    explicit WebSpeechSynthesisClient(WebPage&);
+
     Ref<WebPage> protectedPage() const;
 
     void setObserver(WeakPtr<WebCore::SpeechSynthesisClientObserver> observer) override { m_observer = observer; }
     WeakPtr<WebCore::SpeechSynthesisClientObserver> observer() const override { return m_observer; }
     void resetState() override;
 
     WebCore::SpeechSynthesisClientObserver* corePageObserver() const;
-    
+
     WeakRef<WebPage> m_page;
```

Comment:
Same bug here. The lifetime of the WebSpeechSynthesisClient can now be extended past the WebPage's lifetime, since it has its own RefCount. As a result, this WeakRef may now become null, which is not allowed for a WeakRef. Odds are you'll need a WeakPtr with null checks now.

Reply:
Ah I see what you're saying, that is indeed a problem. In this scenario, could we instead have `WebSpeechSynthesisClient` forward it's ref'ing to `WebPage`?

Reply:
I thought about that but I think WebSpeechSynthesisClient is owned by the Page, not the WebPage.

Reply:
Ok yeah, that makes sense. Since making each of the files in this patch ref-counted is turning out to be a bigger change that I originally thought, I've decided to split it up. I'll handle each of these files in a different patch (unless some of them require really small changes).